### PR TITLE
feat(docs-infra): implement figure styles

### PR DESCRIPTION
**Pull request by sjtrimble**
PR#28396 originally addressed an update via issue #23983 to make images more visible with a white background (implementation of gray "lightbox").

This PR implements those styles defined in PR#28396.

## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Images are often lost since the background change to white.

Issue Number: #23983 | and PR #28396

## What is the new behavior?
Add a shadowed gray 'lightbox' to better define images throughout the documentation.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/src/styles/2-modules/_images.scss [hunk 1]
```diff
@@ -1,4 +1,5 @@
-.content {
+.content,
+.sidenav-content {
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_images.scss]: This shouldn't be necessary 😕
Docs content is wrapped in both `.content` and `.sidenav-content`.

**Review by gkalpak — CHANGES_REQUESTED** [review]
Are there occurrences of `figure > img`, without `.lightbox` (or a `.card`)?
If so, how does one decide when to use `.lightbox` and `.card` and when not?

**Comment by sjtrimble** [comment]
@gkalpak eek :/ It looks like one my fixup commits never made it on there? 

I "THINK" I took care of everything.

The extra lines were there originally and sometimes my editor made it look like I wasn't supposed to indent properly so I thought maybe it would break the markdown if I did...
Screen Shot 2019-10-21 at 1 14 27 PM

<hr>

I was going to update the style guide to reflect when to use the `lightbox` and `card` etc.

File: aio/angular.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
     "packageManager": "yarn",
     "warnings": {
       "typescriptMismatch": false
-    }
+    },
+    "analytics": false
```
Comment (gkalpak) [inline aio/angular.json]: This change looks unrelated to the PR.

File: aio/angular.json [hunk 3]
```diff
@@ -189,4 +190,4 @@
     }
   },
   "defaultProject": "site"
-}
+}
```
Comment (gkalpak) [inline aio/angular.json]: 😢

File: aio/content/guide/dependency-injection-in-action.md [hunk 4]
```diff
@@ -221,8 +227,11 @@ When the property is marked as optional, Angular sets `loggerService` to null an
 
 Here's `HeroBiosAndContactsComponent` in action.
 
-<figure>
-  <img src="generated/images/guide/dependency-injection-in-action/hero-bios-and-contacts.png" alt="Bios with contact into">
+<figure class="lightbox">
+  <div class="card">
+      <img src="generated/images/guide/dependency-injection-in-action/hero-bios-and-contacts.png" alt="Bios with cont
+    act into">
```
Comment (gkalpak) [inline aio/content/guide/dependency-injection-in-action.md]: Unnecessary linebreak.

File: aio/content/guide/dependency-injection-in-action.md [hunk 5]
```diff
@@ -460,8 +475,11 @@ The following example puts `MinimalLogger` to use in a simplified version of `He
 
 The `HeroOfTheMonthComponent` constructor's `logger` parameter is typed as `MinimalLogger`, so only the `logs` and `logInfo` members are visible in a TypeScript-aware editor.
 
-<figure>
-  <img src="generated/images/guide/dependency-injection-in-action/minimal-logger-intellisense.png" alt="MinimalLogger restricted API">
+<figure class="lightbox">
+  <div class="card">
+      <img src="generated/images/guide/dependency-injection-in-action/minimal-logger-intellisense.png" alt="MinimalLog
+    ger restricted API">
```
Comment (gkalpak) [inline aio/content/guide/dependency-injection-in-action.md]: Unnecessary linebreak.

File: aio/content/guide/forms-overview.md [hunk 6]
```diff
@@ -82,9 +84,11 @@ Here's the same component with an input field for a single control implemented u
...
 In template-driven forms, the source of truth is the template.
 
-<figure>
-  <img src="generated/images/guide/forms-overview/key-diff-td-forms.png" alt="Template-driven forms key differences">
-</figure>
+<figure class="lightbox">
+  <div class="card">
+    <img src="generated/images/guide/forms-overview/key-diff-td-forms.png" alt="Template-driven forms key differences">
+  </div>
+  </figure>
```
Comment (gkalpak) [inline aio/content/guide/forms-overview.md]: Incorrect indentation.

File: aio/content/guide/hierarchical-dependency-injection.md [hunk 7]
```diff
@@ -108,8 +108,10 @@ The following diagram represents the relationship between the
 `root` `ModuleInjector` and its parent injectors as the
 previous paragraphs describe.
 
- <figure>
-  <img src="generated/images/guide/dependency-injection/injectors.svg" alt="NullInjector, ModuleInjector, root injector">
+<figure class="lightbox">
+  <div class="card">
+      <img src="generated/images/guide/dependency-injection/injectors.svg" alt="NullInjector, ModuleInjector, root injector">
```
Comment (gkalpak) [inline aio/content/guide/hierarchical-dependency-injection.md]: Incorrect indentation.

File: aio/content/guide/router.md [hunk 8]
```diff
@@ -2129,8 +2147,12 @@ Add some styles to apply when the list item is selected.
...
 When the user navigates from the heroes list to the "Magneta" hero and back, "Magneta" appears selected:
 
-<figure>
-  <img src='generated/images/guide/router/selected-hero.png' alt="Selected List">
+<figure class="lightbox">
+
+  <div class="card">
+    <img src='generated/images/guide/router/selected-hero.png' alt="Selected List">
+  </div>
+
```
Comment (gkalpak) [inline aio/content/guide/router.md]: Unnecessary empty lines 😇

**Comment by gkalpak** [comment]
>> Are there occurrences of `figure > img`, without `.lightbox` (or a `.card`)?
>> If so, how does one decide when to use `.lightbox` and `.card` and when not?
>
> I was going to update the style guide to reflect when to use the `lightbox` and `card` etc.

Were or are? 😕 😛
I definitely agree this should be in the guide.

My concern is that we are complicating the conventions one has to follow, which will inevitably result in them not being consistently followed 😟
I would love if we could simplify things. For example, if `.card` should always be used with `.lightbox`, then we might be able to get rid of the nested `.card` and style the `.lightbox` (or the contained imaged) instead. Or if most `fixure`s will use `.lightbox`, we could make it the default and have another class to "opt-out" of the `.lightbox` styling.

**Comment by sjtrimble** [comment]
@gkalpak Here's what I was thinking on images and figures...

I see there being two types of images (both in documentation and marketing pages) that we utilize regularly:
1) *Aesthetic Image*: Usually inline images that are often there for aesthetic reasons or to reiterate something already written or explained (like this little template hero image  http://localhost:4200/guide/architecture-components#templates-and-views) -> This would have no particular styling
2) *Explanatory Image*: Call out images that further explain the documentation (like the other images on that ex page above) or that show steps (like here http://localhost:4200/start) -> This would always have the lightbox element

Furthermore, from my understanding, `figures` are actually to be used as stand-alone content that you could remove and leave untouched. Some of the images contained in `figures` in the start guide, for example, http://localhost:4200/start#template-syntax are technically not correct because they images follow ":".

To do it "really right" we would almost need to remove all of the `figures` to keep things simple (especially since we are not using captions or image groups anywhere in the doc or don't have a strong need to do so). If we do that (all just `img` tags), then I would probably have a `div.lighbox > img` structure and style accordingly for the explanatory images. What do you think?

For now however .... we could merge this PR (once clean) so it visually looks right, while I update the format again. **Thoughts?**

**Comment by gkalpak** [comment]
Thx for the clarification. SGTM. Happy to get this PR in (once all comments are resolved :wink:).

**Comment by sjtrimble** [comment]
@gkalpak Is there anything missing for this PR?

File: aio/content/guide/dependency-injection-in-action.md [hunk 9]
```diff
@@ -221,8 +227,10 @@ When the property is marked as optional, Angular sets `loggerService` to null an
 
 Here's `HeroBiosAndContactsComponent` in action.
 
-<figure>
-  <img src="generated/images/guide/dependency-injection-in-action/hero-bios-and-contacts.png" alt="Bios with contact into">
+<figure class="lightbox">
+  <div class="card">
+      <img src="generated/images/guide/dependency-injection-in-action/hero-bios-and-contacts.png" alt="Bios with contact into">
```
Comment (gkalpak) [inline aio/content/guide/dependency-injection-in-action.md]:
```suggestion
    <img src="generated/images/guide/dependency-injection-in-action/hero-bios-and-contacts.png" alt="Bios with contact into">
```

File: aio/content/guide/dependency-injection-in-action.md [hunk 10]
```diff
@@ -460,8 +474,10 @@ The following example puts `MinimalLogger` to use in a simplified version of `He
 
 The `HeroOfTheMonthComponent` constructor's `logger` parameter is typed as `MinimalLogger`, so only the `logs` and `logInfo` members are visible in a TypeScript-aware editor.
 
-<figure>
-  <img src="generated/images/guide/dependency-injection-in-action/minimal-logger-intellisense.png" alt="MinimalLogger restricted API">
+<figure class="lightbox">
+  <div class="card">
+      <img src="generated/images/guide/dependency-injection-in-action/minimal-logger-intellisense.png" alt="MinimalLogger restricted API">
```
Comment (gkalpak) [inline aio/content/guide/dependency-injection-in-action.md]:
```suggestion
    <img src="generated/images/guide/dependency-injection-in-action/minimal-logger-intellisense.png" alt="MinimalLogger restricted API">
```

File: aio/content/start/data.md [hunk 11]
```diff
@@ -105,14 +105,18 @@ When the "Buy" button is clicked, you'll use the cart service to add the current
 
 1. To see the new "Buy" button, refresh the application and click on a product's name to display its details.
 
-   <figure>
-     <img src='generated/images/guide/start/product-details-buy.png' alt="Display details for selected product with a Buy button">
-   </figure>
+   <figure class="lightbox">
```
Comment (gkalpak) [inline aio/content/start/data.md]:
```suggestion
    <figure class="lightbox">
```

File: aio/content/start/data.md [hunk 12]
```diff
@@ -105,14 +105,18 @@ When the "Buy" button is clicked, you'll use the cart service to add the current
...
+    <div class="card">
```
Comment (gkalpak) [inline aio/content/start/data.md]:
```suggestion
      <div class="card">
```

File: aio/content/start/data.md [hunk 13]
```diff
@@ -105,14 +105,18 @@ When the "Buy" button is clicked, you'll use the cart service to add the current
...
+      <img src='generated/images/guide/start/product-details-buy.png' alt="Display details for selected product with a Buy button">
```
Comment (gkalpak) [inline aio/content/start/data.md]:
```suggestion
        <img src='generated/images/guide/start/product-details-buy.png' alt="Display details for selected product with a Buy button">
```

File: aio/content/start/data.md [hunk 14]
```diff
@@ -105,14 +105,18 @@ When the "Buy" button is clicked, you'll use the cart service to add the current
...
+    </div>
```
Comment (gkalpak) [inline aio/content/start/data.md]:
```suggestion
      </div>
```

File: aio/content/start/data.md [hunk 15]
```diff
@@ -105,14 +105,18 @@ When the "Buy" button is clicked, you'll use the cart service to add the current
...
+  </figure>
```
Comment (gkalpak) [inline aio/content/start/data.md]:
```suggestion
    </figure>
```

File: aio/content/start/data.md [hunk 16]
```diff
@@ -150,8 +154,10 @@ We'll create the cart page in two steps:
 
     (Note: The "Checkout" button that we provided in the top-bar component was already configured with a `routerLink` for `/cart`.)
 
-    <figure>
+    <figure class="lightbox">
+      <div class="card">
       <img src='generated/images/guide/start/cart-works.png' alt="Display cart page before customizing">
```
Comment (gkalpak) [inline aio/content/start/data.md]:
```suggestion
        <img src='generated/images/guide/start/cart-works.png' alt="Display cart page before customizing">
```

File: aio/content/start/data.md [hunk 17]
```diff
@@ -204,8 +210,10 @@ Services can be used to share data across components:
     1. Click "Checkout" to see the cart.
     1. To add another product, click "My Store" to return to the product list. Repeat the steps above.
 
-    <figure>
+    <figure class="lightbox">
+      <div class="card">
       <img src='generated/images/guide/start/cart-page-full.png' alt="Cart page with products added">
```
Comment (gkalpak) [inline aio/content/start/data.md]:
```suggestion
        <img src='generated/images/guide/start/cart-page-full.png' alt="Cart page with products added">
```

File: aio/content/start/data.md [hunk 18]
```diff
@@ -363,14 +371,18 @@ Now that your app can retrieve shipping data, you'll create a shipping component
 
     Click on the "Checkout" button to see the updated cart. (Remember that changing the app causes the preview to refresh, which empties the cart.)
 
-    <figure>
+    <figure class="lightbox">
+      <div class="card">
       <img src='generated/images/guide/start/cart-empty-with-shipping-prices.png' alt="Cart with link to shipping prices">
```
Comment (gkalpak) [inline aio/content/start/data.md]:
```suggestion
        <img src='generated/images/guide/start/cart-empty-with-shipping-prices.png' alt="Cart with link to shipping prices">
```

File: aio/content/start/data.md [hunk 19]
```diff
@@ -363,14 +371,18 @@ Now that your app can retrieve shipping data, you'll create a shipping component
...
+      </div>
     </figure>
 
     Click on the link to navigate to the shipping prices.
 
-    <figure>
+    <figure class="lightbox">
+      <div class="card">
       <img src='generated/images/guide/start/shipping-prices.png' alt="Display shipping prices">
```
Comment (gkalpak) [inline aio/content/start/data.md]:
```suggestion
        <img src='generated/images/guide/start/shipping-prices.png' alt="Display shipping prices">
```

**Review by gkalpak — APPROVED** [review]

**Comment by gkalpak** [comment]
@sjtrimble, I pushed a fixup commit with some indentation fixes. If you are happy with that, feel free to mark the PR for merging (adding the `PR action: merge` and `PR action: merge-assistance` labels.

**Comment by gkalpak** [comment]
@googlebot I consent.

**Comment by gkalpak** [comment]
**merge-assistance:** Global approval for this docs(-infra)-only change.

**Comment by sjtrimble** [comment]
Thanks @gkalpak , u da best :P

**Pull request closed without merging**